src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -23,7 +23,7 @@ export class SavedObjectsClient {
    * @returns {promise} - { id, type, version, attributes }
   */
   async create(type, attributes = {}, options = {}) {
-    const method = options.overwrite ? 'index' : 'create';
+    const method = get(options, 'overwrite', false) === false && options.id ? 'create' : 'index';
```

Comment:
`index` handles the automatic id generation. `create` is used to fail if there is an id conflict.

Here is a table I created showing when create or index should be used:
![screenshot 2017-06-28 06 40 40](https://user-images.githubusercontent.com/40265/27639972-bd877926-5bcc-11e7-97f3-724641f41582.png)